LibreNMS/Alert/RunAlerts.php

```diff
@@ -551,15 +552,39 @@ public function alertLog($result, $obj, $transport)
         ];
         $prefix[3] = &$prefix[0];
         $prefix[4] = &$prefix[0];
+
+        if ($obj['state'] == 0) {
+            $severity = Alert::OK;
+        } elseif ($obj['state'] == 1) {
+            switch ($obj['severity']) {
```

Comment:
Array lookup is almost always better than switch statements.  (Alert::STATES is an array of all states keyed by name)

```php
$severity = Alert::STATES[$obj['severity']] ??  Alert::UNKNOWN;
```

Reply:
something like this?